my/coding/commits.py

```diff
@@ -2,11 +2,17 @@
 Git commits data for repositories on your filesystem
 """
 
+import shutil
+import string
 from pathlib import Path
 from datetime import datetime, timezone
 from typing import List, NamedTuple, Optional, Dict, Any, Iterator, Set
 
-from ..common import PathIsh, LazyLogger, mcachew
+from ..core.common import PathIsh, LazyLogger, mcachew, Stats
+from ..core.cachew import cache_dir
+from ..core.warnings import high
+
+# TODO: create user_config dataclass?
```

Comment:
Yeah.. I guess as for schema,
- either of emails or names could be optional (but it should warn/throw if both are set to None?)
- for `roots` would be nice to allow to specify a single path, kind of like `Paths/get_files` handling does.. but here the discovery of 'files' is somewhat different, so maybe not worth the trouble and it's fine as a `Sequence[PathIsh]` for now

Reply:
Sound like sensible defaults, sure

I've typically just used email, as I feel theres a non-zero chance that someone else has the same name as me in some giant project that may be cloned in my repos